File: algorithms/1_easy/M/03783_mirror_distance_of_an_integer/solution.cpp
class Solution {
public:
    static int mirrorDistance(const int n) {
        int temp {n};
        int rev_n {0};

        while (temp > 0) {
            rev_n = rev_n * 10 + temp % 10;
            temp /= 10;
        }

        return abs(n - rev_n);
    }
};

File: algorithms/1_easy/M/03783_mirror_distance_of_an_integer/solution.rs
impl Solution {
    pub fn mirror_distance(n: i32) -> i32 {
        let mut temp = n;
        let mut rev_n = 0_i32;

        while temp > 0 {
            rev_n = rev_n * 10 + temp % 10;
            temp /= 10;
        }

        (n - rev_n).abs()
    }
}
